### PR TITLE
Prepare release notes for 8.4 RC1

### PR DESCRIPTION
**Pull request by ljacomet**
Also covers release features.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -84,22 +169,34 @@ val doesntMatch = property.filter { it.contains("baz") } // is empty
...
+```kotlin
+interface Extension {
+    val description: Property<String>
+}
+
+// register "extension" with type Extension
+extension {
+    // Using the set() method call
+    description.set("Hello Property")
+    // Incubating: using lazy property assignment
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md:194]: ❓ Why is this incubating - my understanding is that this is now stable, right?

**Comment by jbartok** [comment]
@bot-gradle cancel

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (jbartok) [reply subprojects/docs/src/docs/release/notes.md:194]: You are right Stefan, thank you. I'll fix it.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -84,22 +169,34 @@ val doesntMatch = property.filter { it.contains("baz") } // is empty
 
 Note that `filter` is evaluated lazily, so the predicate is only evaluated when the value is queried.
```
Comment (ov7a) [inline subprojects/docs/src/docs/release/notes.md:171]:
````suggestion
This feature can be useful for validating external properties, for example:
```kotlin
providers.gradleProperty("someFlag").filter { it != "false" } )
```
The provider above will return value only if the feature is present and does not equal to "false". 
````

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Pull request merged**